### DISCUSSION TITLE
A way to deploy a helm chart from a helm repo with a configmap?

### DISCUSSION THREAD

**Discussion by ckazimie** [argoproj/argo-cd#6370] [Q&A]
Hi,

Many standard helm charts require a configmap to cutomize their behaviour. One example is Nginx Ingress Controller where a custom configmap is used as a source of configuration parameters. 
Is there any way to define an application in Argo CD that points to a chart in a remote helm repo and at the same time defines a config map to be used by the application? I don't means here to override values.yaml but to define your config map to be deployed together with the application.

Best regards
Czeslaw

**Comment by alexmt** [comment]
There is no first-class support of this use case. Argo CD is focused on GitOps use cases only and delegate this type of task to config management tool. You might try to create config management plugin that uses helm to generate manifests and adds missing ConfigMap. E.g. : https://github.com/argoproj/argocd-example-apps/tree/master/plugins/kustomized-helm
